base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1168,20 +1181,28 @@ pub unsafe extern "C" fn encrypted_value_encrypt(
 /// memory leak
 #[no_mangle]
 pub unsafe extern "C" fn encrypted_value_decrypt(
+    wallet: *mut TariWallet,
     encrypted_value: *const TariEncryptedValue,
     error_out: *mut c_int,
 ) -> c_ulonglong {
     let mut error = 0;
     ptr::swap(error_out, &mut error as *mut c_int);
 
+    // TODO: Add macros here
+    if wallet.is_null() {
+        error = LibWalletError::from(InterfaceError::NullError("wallet".to_string())).code;
+        ptr::swap(error_out, &mut error as *mut c_int);
+        return 0;
+    }
+
     if encrypted_value.is_null() {
         error = LibWalletError::from(InterfaceError::NullError("encrypted_value".to_string())).code;
         ptr::swap(error_out, &mut error as *mut c_int);
         return 0;
     }
 
-    let encrypted_value = (*encrypted_value).clone();
-    encrypted_value.todo_decrypt() as c_ulonglong
+    let value = (*wallet).wallet.encryption_service.decrypt_value(&*encrypted_value);
```

Comment:
This method will throw an error if the value cannot be decrypted, which must be handled here